File: stellara-gallery-backend/src/main/java/com/xmz/stellaragallerybackend/model/entity/Picture.java
package com.xmz.stellaragallerybackend.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("picture")
/**
 * 图片实体。
 */
public class Picture implements Serializable {

    /**
     * 图片主键 ID。
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 原图访问地址。
     */
    private String url;

    /**
     * 缩略图访问地址。
     */
    private String thumbnailUrl;

    /**
     * 图片名称。
     */
    private String name;

    /**
     * 图片简介。
     */
    private String introduction;

    /**
     * 分类 ID。
     */
    private Long categoryId;

    /**
     * 标签 JSON。
     */
    private String tags;

    /**
     * 文件大小，单位字节。
     */
    private Long picSize;

    /**
     * 图片宽度。
     */
    private Integer picWidth;

    /**
     * 图片高度。
     */
    private Integer picHeight;

    /**
     * 图片宽高比。
     */
    private Double picScale;

    /**
     * 图片格式。
     */
    private String picFormat;

    /**
     * 主色调，本期先预留。
     */
    private String picColor;

    /**
     * 上传用户 ID。
     */
    private Long userId;

    /**
     * 所属空间 ID。
     */
    private Long spaceId;

    /**
     * 是否公开：0 私有，1 公开。
     */
    private Integer isPublic;

    /**
     * 审核状态：0 待审核，1 通过，2 拒绝。
     */
    private Integer reviewStatus;

    /**
     * 审核说明。
     */
    private String reviewMessage;

    /**
     * 审核人 ID。
     */
    private Long reviewerId;

    /**
     * 审核时间。
     */
    private LocalDateTime reviewTime;

    /**
     * 去重浏览量 UV，来源于 Redis HyperLogLog。
     */
    private Integer viewCount;

    /**
     * 点赞数。
     */
    private Integer likeCount;

    /**
     * 收藏数。
     */
    private Integer favoriteCount;

    /**
     * 下载次数。
     */
    private Integer downloadCount;

    /**
     * 创建时间。
     */
    private LocalDateTime createTime;

    /**
     * 更新时间。
     */
    private LocalDateTime updateTime;

    /**
     * 逻辑删除标记。
     */
    @TableLogic
    @TableField("is_delete")
    private Integer isDelete;
}

File: stellara-gallery-frontend/src/api/picture.ts
import { API_BASE, request, type BaseResponse, type PageResult } from './user'

export interface PictureVO {
  id: number
  url: string
  thumbnailUrl?: string
  name: string
  introduction?: string
  categoryId?: number
  categoryName?: string
  tags?: string[]
  picSize?: number
  picWidth?: number
  picHeight?: number
  picScale?: number
  picFormat?: string
  picColor?: string
  userId: number
  userName?: string
  userAvatar?: string
  spaceId?: number
  isPublic: number
  reviewStatus: number
  reviewMessage?: string
  reviewerId?: number
  reviewTime?: string
  viewCount: number
  likeCount: number
  favoriteCount: number
  downloadCount: number
  liked?: boolean
  favorited?: boolean
  createTime?: string
  updateTime?: string
}

export interface PictureQueryRequest {
  current?: number
  pageSize?: number
  sortField?: string
  sortOrder?: 'ascend' | 'descend'
  keyword?: string
  categoryId?: number
  tag?: string
  userId?: number
  spaceId?: number
  isPublic?: number
  reviewStatus?: number
}

export interface PictureUpdateRequest {
  id: number
  name: string
  introduction?: string
  categoryId?: number
  tags?: string[]
  isPublic?: number
}

export interface PictureReviewRequest {
  id: number
  reviewStatus: number
  reviewMessage?: string
}

export interface PictureUploadForm {
  file: File
  name: string
  introduction?: string
  categoryId?: number
  tags?: string[]
  isPublic?: number
}

async function uploadRequest<T>(path: string, formData: FormData): Promise<T> {
  const response = await fetch(`${API_BASE}${path}`, {
    method: 'POST',
    credentials: 'include',
    body: formData,
  })
  const result = (await response.json()) as BaseResponse<T>
  if (!response.ok || result.code !== 0) {
    throw new Error(result.message || '请求失败')
  }
  return result.data
}

export function uploadPicture(data: PictureUploadForm) {
  const formData = new FormData()
  formData.append('file', data.file)
  formData.append('name', data.name)
  formData.append('introduction', data.introduction ?? '')
  if (data.categoryId) {
    formData.append('categoryId', String(data.categoryId))
  }
  formData.append('tags', JSON.stringify(data.tags ?? []))
  formData.append('isPublic', String(data.isPublic ?? 0))
  return uploadRequest<PictureVO>('/picture/upload', formData)
}

export function getPictureById(id: number) {
  return request<PictureVO>(`/picture/get?id=${encodeURIComponent(id)}`)
}

export function listPublicPictures(data: PictureQueryRequest) {
  return request<PageResult<PictureVO>>('/picture/list/public/page', {
    method: 'POST',
    body: JSON.stringify(data),
  })
}

export function listMyPictures(data: PictureQueryRequest) {
  return request<PageResult<PictureVO>>('/picture/list/my/page', {
    method: 'POST',
    body: JSON.stringify(data),
  })
}

export function listAdminPictures(data: PictureQueryRequest) {
  return request<PageResult<PictureVO>>('/picture/list/admin/page', {
    method: 'POST',
    body: JSON.stringify(data),
  })
}

export function listFavoritePictures(data: PictureQueryRequest) {
  return request<PageResult<PictureVO>>('/picture/favorite/list/page', {
    method: 'POST',
    body: JSON.stringify(data),
  })
}

export function updatePicture(data: PictureUpdateRequest) {
  return request<boolean>('/picture/update', {
    method: 'POST',
    body: JSON.stringify(data),
  })
}

export function deletePicture(id: number) {
  return request<boolean>('/picture/delete', {
    method: 'POST',
    body: JSON.stringify({ id }),
  })
}

export function reviewPicture(data: PictureReviewRequest) {
  return request<boolean>('/picture/review', {
    method: 'POST',
    body: JSON.stringify(data),
  })
}

export function batchReviewPicture(ids: number[], reviewStatus: number, reviewMessage?: string) {
  return request<boolean>('/picture/review/batch', {
    method: 'POST',
    body: JSON.stringify({ ids, reviewStatus, reviewMessage }),
  })
}

export function toggleLike(id: number) {
  return request<boolean>('/picture/like', {
    method: 'POST',
    body: JSON.stringify({ id }),
  })
}

export function toggleFavorite(id: number) {
  return request<boolean>('/picture/favorite', {
    method: 'POST',
    body: JSON.stringify({ id }),
  })
}

export function downloadPicture(id: number) {
  return request<string>(`/picture/download?id=${encodeURIComponent(id)}`)
}
